src/protocol/mod.rs

```diff
@@ -148,29 +148,36 @@ macro_rules! gen_rep {
 
                     let rep = $mod::Rep::load(&buf).map_err(|e| ProtocolError::new_err(e.to_string()))?;
 
-                    let ret = unsafe {
-                        match rep {
-                            rep @ $mod::Rep::Ok $({ $($tt)+ })? => {
-                                let initializer =
-                                    ::pyo3::PyClassInitializer::from(([<$base_class Ok>], Self(rep)));
+                    let ret = match rep {
+                        rep @ $mod::Rep::Ok $({ $($tt)+ })? => {
+                            let initializer =
+                                ::pyo3::PyClassInitializer::from(([<$base_class Ok>], Self(rep)));
+                            // SAFETY: We use the same when creating the `initializer` and creating a new object.
```

Comment:
I don't understand this comment, can you rephrase it ? (or provide more information in the current discussion)

Reply:
I've rephrase the comment, WDYT ?